tests/test_parquet.py

```diff
@@ -113,5 +131,76 @@ def testMultiIndexDataFrame(self):
             self.butler.get(self.datasetType, dataId={}, parameters={"columns": ["d"]})
 
 
+@unittest.skipUnless(pd is not None, "Cannot test DataFrameDelegate without pandas.")
+class DataFrameDelegateTestCase(unittest.TestCase):
```

Comment:
If you have an in-memory datastore test case you might be able to remove a lot of the code duplication here and reserve this test case class for dealing with all the places that trigger exceptions.